### PR TITLE
ARROW-9368: [Python] Rename predicate argument to filter in split_by_row_group()

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
Small change, for consistency with the `filter` keyword in `to_table()` and `get_fragments()` methods.

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
I thought to rename the C++ SplitByRowGroup as well, but apparently on the C++ side we are consistently using `predicate` (`GetFragments` also uses predicate), so for now left it as is.

File: python/pyarrow/_dataset.pyx [hunk 1]
```diff
@@ -936,7 +936,7 @@ cdef class ParquetFileFragment(FileFragment):
 
         Parameters
         ----------
-        predicate : Expression, default None
+        filter : Expression, default None
             Exclude RowGroups whose statistics contradicts the predicate.
```
Comment (pitrou) [inline python/pyarrow/_dataset.pyx:940]: Update this sentence? Perhaps "Only include those row groups which satisfy this predicate"

**Review by bkietz — APPROVED** [review]
LGTM

**Comment by pitrou** [comment]
Thank you @jorisvandenbossche !

**Pull request closed without merging**